include/private/tdydiscretizationimpl.h

```diff
@@ -0,0 +1,32 @@
+#if !defined(TDYDISCRETIZATION_H)
+#define TDYDISCRETIZATION_H
+
+#include <petsc.h>
+#include <private/tdydmimpl.h>
+
+
```

Comment:
@bishtgautam , would you mind adding a brief comment here describing how we're following PFLOTRAN's lead here, and that we may elect to use DMShell down the line?

Reply:
Fixed.